src/OrleansRuntime/MembershipService/GrainBasedMembershipTable.cs

```diff
@@ -32,6 +34,16 @@ internal class GrainBasedMembershipTable : Grain, IMembershipTable
         private InMemoryMembershipTable table;
         private TraceLogger logger;
 
+        public Task InitializeMembershipTable(GlobalConfiguration config, bool tryInitTableVersion, TraceLogger traceLogger)
+        {
+            throw new InvalidOperationException("This method should not be called directly");
```

Comment:
my current code actually checks for that specifically:
```c
GlobalConfiguration.LivenessProviderType livenessType = config.LivenessType;
            if (livenessType.Equals(GlobalConfiguration.LivenessProviderType.MembershipTableGrain))
            {
                membershipTable =
                    GrainFactory.Cast<IMembershipTable>(GrainReference.FromGrainId(Constants.SystemMembershipTableId));
            }
            else
            {
```
i did that because calling InitializeMembershipTable on GrainBasedMembershipTable in the context of MembershipFactory throws 
```c
throw new InvalidExpressionException(
                    String.Format("Trying to send a message on a silo not from within grain and not from within system target (RuntimeContext is not set to SchedulingContext) "
                        + "RuntimeContext.Current={0} TaskScheduler.Current={1}",
                        RuntimeContext.Current == null ? "null" : RuntimeContext.Current.ToString(),
                        TaskScheduler.Current));
```
anyway, to preserve consistency, maybe this is a better solution:
```c
public Task InitializeMembershipTable(GlobalConfiguration config, bool tryInitTableVersion, TraceLogger traceLogger)
        {
            return TaskDone.Done;
        }

        public Task DeleteMembershipTableEntries(string deploymentId)
        {
            table = new InMemoryMembershipTable();
            return TaskDone.Done;
        }
```

also, should DeleteMembershipTableEntries delete all rows but the deployment row itself? (the ZK implementation currently deletes the deployment itself with all rows). I think that it should be named DeleteMembershipTable and get a bool whether to delete the deployment itself or not. this will make the method more testable and easy to understand.